src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -152,6 +154,13 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 			connectionManager = PoolingHttpClientConnectionManager().apply {
 				defaultMaxPerRoute = (maxConnections * 0.9).toInt()
 				maxTotal = maxConnections
+				logger.debug("Initial maxConnections set is ${maxConnections}")
+				Settings.CONNECTIONSPEED.listen { newValue ->
+					logger.debug("Changed maxConnections to ${maxConnections}")
+					maxConnections = min(Settings.CONNECTIONSPEED.get(), Runtime.getRuntime().availableProcessors().coerceAtLeast(2) * 50)
```

Comment:
This code is used twice, please add a little helper function. Also, you already have the `newValue`, why are you calling `Settings.CONNECTIONSPEED.get()`?

Reply:
No idea why, thanks for pointing it out

Reply:
And what code is used twice ? The debug, or the listen ?

Reply:
Or maybe the min() ?

Reply:
the min

Reply:
maybe just something like `updateMaxConnections(newValue)`

Reply:
This ends up being less clear imo; here it is : 51ab6dd529ec7da269a3ba71277743619c59a09d